src/main/java/com/cbms/rul/models/LinearRegressionModelImpl.java

```diff
@@ -16,33 +16,40 @@
 
 public class LinearRegressionModelImpl implements ModelStrategy {
     /**
-     This function takes the filtered training dataset and trains a linear regression regression model,
-     after that it returns the model.
-     To use this method you need to pass the training dataset.
-     @author Talal
+     * This function takes the filtered training dataset and trains a linear regression regression model,
+     * after that it returns the model.
+     * To use this method you need to pass the training dataset.
+     *
+     * @author Talal
      */
     @Override
-    public Classifier trainModel(Instances firstTrain) throws Exception {
-        firstTrain.setClassIndex(firstTrain.numAttributes()-1);
+    public Classifier trainModel(Instances firstTrain) {
+        firstTrain.setClassIndex(firstTrain.numAttributes() - 1);
         Instances trainDataset = removeInstances(firstTrain);
         //SMOreg lr = new SMOreg();
-        LinearRegression lr=new LinearRegression();
-        lr.buildClassifier(firstTrain);
+        LinearRegression lr = new LinearRegression();
+        try {
+            lr.buildClassifier(firstTrain);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
         return lr;
     }
 
     /**
-     This function removes the outliers (data that can affect the prediction of the model)
-     from a training dataset.
-     After outliers are removed, new dataset is returned.
-     This function is used to improve the performance of a model.
-     To use this function you need to pass the training dataset as a parameter.
-     @author Talal
+     * This function removes the outliers (data that can affect the prediction of the model)
+     * from a training dataset.
+     * After outliers are removed, new dataset is returned.
+     * This function is used to improve the performance of a model.
+     * To use this function you need to pass the training dataset as a parameter.
+     *
+     * @author Talal
      */
-    private Instances removeInstances (Instances trainDataset) {
-        for (int i = 0; i < trainDataset.numInstances() ; i++) {
+    private Instances removeInstances(Instances trainDataset) {
+        for (int i = 0; i < trainDataset.numInstances(); i++) {
             Instance inst = trainDataset.instance(i);
-            if (inst.value(inst.classAttribute())>150) {
+            if (inst.value(inst.classAttribute()) > 150) {
```

Comment:
We have to change this 150 value.

Reply:
yes, it will be done when we get to the parameter page where the user will be able to customize his settings, probably next sprint

Reply:
Ok perfect.